### ISSUE TITLE
Consider dropping gcc travis build for OS X

### ISSUE THREAD

**Issue by mp15** [samtools/samtools#506]
It's just a wrapper for clang anyway.

````
$ export CC=gcc
$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
````

**Comment by bewt85** [comment]
Done in https://github.com/samtools/samtools/pull/527

**Comment by jmarshall** [comment]
Fixed by #527.
